BaseTools/Scripts/GitHub.py

```diff
@@ -0,0 +1,160 @@
+## @file
+#  GitHub API helper functions.
+#
+#  Copyright (c) Microsoft Corporation.
+#  SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+
+import logging
+import re
+import requests

```

Comment:
If the github module is used, it can simplify the code and remove the need for requests.

Using the requests method can run into limits on the number of requests per hour where using github module with authenticated token does not.

Perhaps this is a good quick solution, but needs to be followed up with a more scalable one soon.

Reply:
The `GITHUB_TOKEN` can support [1,000 requests per hour](https://docs.github.com/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-github_token-in-github-actions). If I understand the app you're talking about correctly, that will support 5,000 requests per hour.

Over the course of the last week, 44 pull requests have been active, so I don't see that limit as particularly concerning. Even if all of those pull requests had happened in one hour and each received 20 pushes, that would only be 880 requests for the hour.

<img width="497" alt="image" src="https://github.com/user-attachments/assets/5a0ab151-df7a-4543-bc7b-e07d64090034">

I agree a higher rate limit is better, but I just want to state that based on historical data this shouldn't be a practical issue for the repo.

Reply:
Thanks for the analysis.  I agree this can be a solution now and improve later.